x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -1004,6 +1100,14 @@ export class DataRecognizer {
       }
     });
 
+    if (generalOverrides.some(override => !!override.analysis_limits?.model_memory_limit)) {
+      this.jobsForModelMemoryEstimation = [];
+    } else {
+      this.jobsForModelMemoryEstimation = jobSpecificOverrides
```

Comment:
i don't think this logic is correct. we'd want to be looping over all jobs in the module, and filtering out the ones that are listed in `jobSpecificOverrides` which also have `model_memory_limit` set

Reply:
Fix with the latest commits